meresco/dans/longconverter.py

```diff
@@ -1311,7 +1311,8 @@ def _getFunding(self, lxmlNode, e_longmetadata):
                     etree.SubElement(e_ga, "description").text = GaDescription[0]                   
                 GaFunderId = ga.xpath('self::gal:grantAgreement/gal:funder/@IDref', namespaces=namespacesmap)
                 if GaFunderId:
-                    #found Funder: Get its (plain) name TODO: Check for nameIdentifier and add it to 
+                    etree.SubElement(e_ga, "funderIdentifier", type="").text = GaFunderId[0]
```

Comment:
De `@IDref `is GEEN nameIdentifier. Dus deze als zodanig toevoegen is niet correct. De `@IDref` verwijst naar een `<mods:name ID="">` element, waaraan wél een nameIdentifier gekoppeld zou kunnen zijn. Deze dient dus opgehaald te worden via `ID` en, indien aanwezig, hier toegevoegd als "funderIdentifier", met Type aanduiding dus ook. We willen hier enkel valide nameIdentifiers zien, anders heeft het geen zin (kan niet op gezocht worden dan).
Dus uit onderstaande zou dus de ROR als funderIdentifier gevonden dienen te worden.:
```
<name ID="FND398221_2" type="corporate">
    <namePart>European Commission</namePart>
    <nameIdentifier type="ror">00k4n6c32</nameIdentifier>
    <role>
        <roleTerm authority="marcrelator" type="code">fnd</roleTerm>
    </role>
</name>
 (...)
<gal:grantAgreement code="info:eu-repo/grantAgreement/EC/FP7/23456">
	<gal:funder IDref="FND398221_2"/>
</gal:grantAgreement>
```

Moet dus zoiets worden:
`<funderIdentifier", type="ror">00k4n6c32</funderIdentifier>`
En niet:
`<funderIdentifier", type="">FND398221_2</funderIdentifier>`

Reply:
Er kan ook nog een DAI gekoppeld zijn aan een name element via de [mods-dai-extentie](https://harvester.dans.knaw.nl/). Vergeet deze maar, want we willen van de dai-extentie af en is depricated...

Reply:
Ok, ik snap het.  Ik heb het nu veranderd.
Het is nu wel zo (dezelfde manier als in `Datacite`) dat als de type van de nameidentifier wordt niet herkend, wordt het nog steeds als een valide identifier gezien:

  ```
if funderId.is_valid():
      etree.SubElement(e_ga, "funderIdentifier", type=funderId.get_name()).text = funderId.get_id()
```

In `NameIdentifierFactory` wordt het als `Unknown` type nameidentifer gezien.

B.v. als de type `onzin` is en de identifier `00k4n6c32`, kirijg je in knaw_long een nameidentifier `onzin:00k4n6c32`.

Zoals gezegd, gaat het in Datacite dezelfde manier.

Reply:
Ah, bedoel je dat de method funderId.is_valid() True geeft indien unknown?

Reply:
yep.

`class Unknown(NameIdentifier):`
    ID_PATTERN = None

    def __init__(self, type, baseDigits):
        NameIdentifier.__init__(self, type, baseDigits)

    def validate_and_initialize(self):

        if self.init_value.lower().startswith(self.name.lower() + ":"):
            self.unformatted = self.init_value[len(self.name) + 1:]
            self.id = '%s:%s' % (self.name.lower(), self.unformatted)
        else:
            self.unformatted = self.init_value
            self.id = self.name.lower() + ":" + self.init_value

        self.typedVariants.add(self.unformatted)
        self.typedVariants.add(self.id)

        self.valid = True  # Set to true, for we have no knowledge of validating this identifier, but we want to use it anyway.

    def __str__(self):
        return self.get_init_value()